hawkular-wildfly-agent-installer/src/main/java/org/hawkular/wildfly/agent/installer/AgentInstaller.java

```diff
@@ -349,8 +352,21 @@ private static String createSecurityRealm(String keystoreFile, String keystorePa
      */
     private static XmlEdit createStorageAdapter(boolean withHttps, InstallerConfiguration installerConfig) {
         String select = "/server/profile/*[namespace-uri()='urn:org.hawkular.agent:agent:1.0']/";
-        StringBuilder xml = new StringBuilder("<storage-adapter")
-                .append(" type=\"HAWKULAR\"");
+        StringBuilder xml = new StringBuilder("<storage-adapter");
+
+        String tenantId = installerConfig.getTenantId();
+        if (installerConfig.isMetricsOnlyMode()) {
+            xml.append(" type=\"METRICS\"");
+            if (tenantId == null || tenantId.isEmpty()) {
+                throw new IllegalArgumentException("You must specify tenant-id when in metrics-only mode");
+            }
+        } else {
+            xml.append(" type=\"HAWKULAR\"");
+        }
+
+        if (tenantId != null && !tenantId.isEmpty()) {
+            xml.append(" tenantId=\"" + tenantId + "\"");
```

Comment:
I see no problem with that, if it acts the way you said. My only concern would be that the agent could send bad data to the backend, but it doesn't seem to be the case then.